### PR TITLE
added firefox developer support

### PR DESCRIPTION
**Pull request by xKiian**
[#555](https://github.com/httptoolkit/httptoolkit/issues/555)

### CODE REVIEW

File: src/interceptors/fresh-firefox.ts [hunk 1]
```diff
@@ -41,13 +40,13 @@ const testCertutil = (command: string, options?: SpawnOptions) => {
 
 const getCertutilCommand = _.memoize(async () => {
     // If a working certutil is available in our path, we're all good
-    if (await testCertutil('certutil')) return { command: 'certutil' };
+    if (await testCertutil('certutil')) return {command: 'certutil'};
```
Comment (pimterry) [inline src/interceptors/fresh-firefox.ts:43]: Can you undo all the formatting changes from this file? It makes this quite difficult to review, and it breaks `git blame` later on.

**Review by pimterry — CHANGES_REQUESTED** [review]
Looks good in general (although it would be clearer without the other changes). Thanks for this PR, great to see this making progress! :tada:

The build is failing though, just because there's an e2e integration test [here](https://github.com/httptoolkit/httptoolkit-server/blob/1f345cba6693540f31eae54a580ed3f614b2b6c9/test/integration/e2e-api-test.spec.ts#L253-L274) that checks the specific interceptors the API returns. You'll just need to add the two new firefox entries there as new `inactivable` options in the right place (inactivable because these aren't installed in GitHub's CI by default, which is the benchmark target env we use for those e2e tests).

Where have you tested this setup btw? It would be useful to know which OSs and OS versions you've used - it should be the same for all of them, but detection particularly has a few platform specific bits, so I'll need to test any cases you haven't covered before we ship this.

File: src/interceptors/fresh-firefox.ts (hunk 1 above)
Comment (xKiian) [reply src/interceptors/fresh-firefox.ts:43]: oh sorry, yea sure

**Comment by xKiian** [comment]
i can only test the changes on windows 11 sadly

**Comment by pimterry** [comment]
> i can only test the changes on windows 11 sadly

No problem, that's fine! It'll need to wait until I'm back in my office next week, but I can properly test both Linux & Mac there no problem, so that will cover everything. Once all the other changes are done here and the tests are passing I think this is OK to merge anyway, any detection issues can be fixed elsewhere later.

**Comment by xKiian** [comment]
> but detection particularly has a few platform specific bits, so I'll need to test any cases you haven't covered before we ship this.

I didn't change the functionality for the detection.

**Comment by pimterry** [comment]
> I didn't change the functionality for the detection.

Yep, understood. The only concern is because the detection for these browser variants hasn't been used until now, so it hasn't been thoroughly tested, it's not because it's changed at all.

**Review by pimterry — APPROVED** [review]

**Comment by pimterry** [comment]
Now merged! :+1: Awesome work, thanks for contributing this. I've tested in Linux already and it's working well. It'll have to wait until I'm back next week to test and confirm on Mac, but I expect that'll be fine, and then this is good to go.

I'm not sure if you're aware, but HTTP Toolkit Pro is free for contributors - just let me know your email and I'll set you up with an account.

**Comment by xKiian** [comment]
Thanks, but i dont feel comfortable sharing my e-mail here. Can i send you an email?

**Comment by pimterry** [comment]
Sure, I'm tim @ httptoolkit.com

**Comment by xKiian** [comment]
Thanks!

**Pull request merged**